### ISSUE TITLE
./run: line 84: syntax error: unexpected "("

### ISSUE THREAD

**Issue by yonas** [disaster37/alpine-gocd-server#2]
Launched using Rancher (found it in the catalog). The container continually fails and restarts.

Image: `webcenter/alpine-gocd-server:17.3.0-1`

```
6/17/2017 1:50:46 PM
6/17/2017 1:50:46 PM                                                           .                           `.
6/17/2017 1:50:46 PM                                                          -d+                          +d.
6/17/2017 1:50:46 PM                 `......................................  yms........----------.       hms
6/17/2017 1:50:46 PM                :hdddddddddddddddddddddddddddddddddddddd:`dmmdddddddddmmmmmmmmddo     `hmm`
6/17/2017 1:50:46 PM              `ommmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmm-:mmmmmmmmmmmmmmmmmmmmmmm-.::+hmmm-
6/17/2017 1:50:46 PM             .ymmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmd /mmmmmmmmmmmmmmmmmmmmmmm-+mmmmmmm/
6/17/2017 1:50:46 PM            :hmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmh``/ooooooshmmmmmmmmmmmmmm-:oooooo/
6/17/2017 1:50:46 PM           /dmmmdmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmhssssssssydmmmmmmmmmmmmmm-
6/17/2017 1:50:46 PM          ommdy//mmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmm-
6/17/2017 1:50:46 PM          -so-  :mmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmdsmmmmmmmmmmmmmm-
6/17/2017 1:50:46 PM                :mmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmm+:mmmmmmmmmmmmmm-
6/17/2017 1:50:46 PM                :mmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmm/:mmmmmmmmmmmmmm-
6/17/2017 1:50:46 PM                :mmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmm/:mmmmmmmmmmmmmm-
6/17/2017 1:50:46 PM                :mmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmm+`sdmmmmmmmmmmdo
6/17/2017 1:50:46 PM                :mmmmmmmmmmmmmd+--------------------------:smmmmmm+..----------`
6/17/2017 1:50:46 PM                :mmmmmmmmmmmmmo                            `mmmmmmmmmmmmmd
6/17/2017 1:50:46 PM                :mmmmmmmmmmmmmo                            `mmmmmmmmmmmmmd
6/17/2017 1:50:46 PM                :mmmmmmmmmmmmmo                            `mmmmmmmmmmmmmd
6/17/2017 1:50:46 PM                :mmmmmmmmmmmmmo                            `mmmmmmmmmmmmmd
6/17/2017 1:50:46 PM                .hmmmmmmmmmmmd/                             ymmmmmmmmmmmms
6/17/2017 1:50:46 PM                 `````````````                               ````````````
6/17/2017 1:50:46 PM
6/17/2017 1:50:46 PM                Name: alpine-gocd-server
6/17/2017 1:50:46 PM                Site: https://www.gocd.io
6/17/2017 1:50:46 PM                Image Author: Sebastien LANGOUREAUX <linuxworkgroup@hotmail.com>
6/17/2017 1:50:46 PM                Image support: https://github.com/disaster37/alpine-gocd-server/issues
6/17/2017 1:50:46 PM                Rancher: http://rancher.com
6/17/2017 1:50:46 PM                Try Rancher: https://try.rancher.com/
6/17/2017 1:50:46 PM
6/17/2017 1:50:46 PM      For french people, you can visit my blog http://blog.webcenter.fr about Docker and Rancher
6/17/2017 1:50:46 PM
6/17/2017 1:50:46 PM
6/17/2017 1:50:46 PM      .ooooooo+:`     /ooo-    /ooo:    +oo   -/osyso+- `+oo:    +oo: +ooooooooo`ooooooo+:`
6/17/2017 1:50:46 PM      -mmmhyydmmh-   -mmmmh`   smmmm:   dmm``sdmdyosdmdo`dmmo    hmmo dmmmyyyyyy.mmmhyydmmh-
6/17/2017 1:50:46 PM      -mmm/  -mmmo  `ymdhmmo   smmmmd/  dmm`ymmm-   /mdm.dmmo    hmmo dmmh      -mmm/  -mmmo
6/17/2017 1:50:46 PM      -mmms//ymmh-  +mmo:mmm-  smmsmmm/ dmd`mmmy    `--.`dmmhssssmmmo dmmdoooo+ -mmms//smmh-
6/17/2017 1:50:46 PM      -mmmddmmm/`  .mmd.`ymmh` smm-/mmm:hmd`mmms        `dmmdyyyymmmo dmmdhhhhy -mmmddmmm+`
6/17/2017 1:50:46 PM      -mmm+.smmh` `ymmdhhhmmmo smm- /dmdmmm`hmmd`   `sss-dmmo    hmmo dmmh      -mmm+.smmh.
6/17/2017 1:50:46 PM      -mmm/ `ymmh/smmy////ommm-smm-  :dmmmm`-hmmy/::smmh`dmmo    hmmo dmmd/////+.mmm/ `ymmh+-
6/17/2017 1:50:46 PM      .hhh/  `ohhhhhh.     yhhsshd-   :hhhh` .+yhdddhy+``hhh+    yhh+ yhhhhhhhhd.hhh/  `ohhd:
6/17/2017 1:50:46 PM       ```      ````        ``` ``      ```     `..``     ``      ``   ````````` ```
6/17/2017 1:50:46 PM       Community ^_^
6/17/2017 1:50:46 PM
6/17/2017 1:50:46 PM[cont-init.d] 100-welcome.sh: exited 0.
6/17/2017 1:50:46 PM[cont-init.d] 101-gen-confd-config.sh: executing...
6/17/2017 1:50:46 PM[cont-init.d] 101-gen-confd-config.sh: exited 0.
6/17/2017 1:50:46 PM[cont-init.d] 102-init-confd.sh: executing...
6/17/2017 1:50:46 PM2017-06-17T17:50:46Z gocd-gocd-server-1 /opt/confd/bin/confd[385]: INFO Backend set to env
6/17/2017 1:50:46 PM2017-06-17T17:50:46Z gocd-gocd-server-1 /opt/confd/bin/confd[385]: INFO Starting confd
6/17/2017 1:50:46 PM2017-06-17T17:50:46Z gocd-gocd-server-1 /opt/confd/bin/confd[385]: INFO Backend nodes set to
6/17/2017 1:50:46 PM2017-06-17T17:50:46Z gocd-gocd-server-1 /opt/confd/bin/confd[385]: INFO /etc/services.d/gocd/run has mode -rwxr-xr-x should be -rwxr--r--
6/17/2017 1:50:46 PM2017-06-17T17:50:46Z gocd-gocd-server-1 /opt/confd/bin/confd[385]: INFO Target config /etc/services.d/gocd/run out of sync
6/17/2017 1:50:46 PM2017-06-17T17:50:46Z gocd-gocd-server-1 /opt/confd/bin/confd[385]: INFO Target config /etc/services.d/gocd/run has been updated
6/17/2017 1:50:46 PM[cont-init.d] 102-init-confd.sh: exited 0.
6/17/2017 1:50:46 PM[cont-init.d] done.
6/17/2017 1:50:46 PM[services.d] starting services
6/17/2017 1:50:46 PM[services.d] done.
6/17/2017 1:50:46 PM./run: line 84: syntax error: unexpected "("
6/17/2017 1:50:46 PM[cont-finish.d] executing container finish scripts...
6/17/2017 1:50:46 PM[cont-finish.d] done.
6/17/2017 1:50:46 PM[s6-finish] syncing disks.
6/17/2017 1:50:46 PM[s6-finish] sending all processes the TERM signal.
6/17/2017 1:50:49 PM[s6-finish] sending all processes the KILL signal and exiting.
```

**Comment by disaster37** [comment]
Hi,

can you try with following image `webcenter/alpine-gocd-agent:17.3.0-2` ?
If you are the same error, can you paste the docker-compose and rancher-compose without any password ;)

**Comment by yonas** [comment]
Thanks @disaster37, it didn't seem to work. Here is the docker-compose and rancher-compose:

### Docker Compose
```
version: '2'
volumes:
  gocd-server-data:
    driver: local
  gocd-agent-data:
    driver: local
    per_container: true
  gocd-scheduler-setting:
    driver: local
    per_container: true
services:
  docker-engine:
    privileged: true
    image: index.docker.io/docker:1.13-dind
    volumes:
    - gocd-agent-data:/data
    links:
    - gocd-server:gocd-server
    labels:
      io.rancher.scheduler.affinity:container_label_soft_ne: io.rancher.stack_service.name=$${stack_name}/$${service_name}
      io.rancher.container.hostname_override: container_name
  rancher-cattle-metadata:
    image: webcenter/rancher-cattle-metadata:1.0.1
    network_mode: none
    volumes:
    - gocd-scheduler-setting:/opt/scheduler
    links:
    - gocd-server:gocd-server
    labels:
      io.rancher.container.start_once: 'true'
      io.rancher.scheduler.affinity:container_label_soft_ne: io.rancher.stack_service.name=$${stack_name}/$${service_name}
      io.rancher.container.hostname_override: container_name
  gocd-server:
    image: webcenter/alpine-gocd-server:17.3.0-1
    environment:
      CONFD_BACKEND: env
      CONFD_NODES: ''
      CONFD_PREFIX_KEY: /gocd
      GOCD_CONFIG_agent-key: uE%th80B,Q0|4VPK
      GOCD_CONFIG_memory: 1024m
      GOCD_CONFIG_server-url: https://gocd-server:8154/go
      GOCD_PLUGIN_docker-pipline: https://github.com/Haufe-Lexware/gocd-plugins/releases/download/v1.0.0-beta/gocd-docker-pipeline-plugin-1.0.0.jar
      GOCD_PLUGIN_docker-task: https://github.com/manojlds/gocd-docker/releases/download/0.1.27/docker-task-assembly-0.1.27.jar
      GOCD_PLUGIN_email-notifier: https://github.com/gocd-contrib/email-notifier/releases/download/v0.1/email-notifier-0.1.jar
      GOCD_PLUGIN_github-auth: https://github.com/gocd-contrib/gocd-oauth-login/releases/download/v2.3/github-oauth-login-2.3.jar
      GOCD_PLUGIN_github-notifier: https://github.com/gocd-contrib/gocd-build-status-notifier/releases/download/1.3/github-pr-status-1.3.jar
      GOCD_PLUGIN_github-scm: https://github.com/ashwanthkumar/gocd-build-github-pull-requests/releases/download/v1.3.3/github-pr-poller-1.3.3.jar
      GOCD_PLUGIN_gitlab-auth: https://github.com/gocd-contrib/gocd-oauth-login/releases/download/v2.3/gitlab-oauth-login-2.3.jar
      GOCD_PLUGIN_google-auth: https://github.com/gocd-contrib/gocd-oauth-login/releases/download/v2.3/google-oauth-login-2.3.jar
      GOCD_PLUGIN_maven-repository: https://github.com/1and1/go-maven-poller/releases/download/v1.1.4/go-maven-poller.jar
      GOCD_PLUGIN_maven-task: https://github.com/ruckc/gocd-maven-plugin/releases/download/0.1.1/gocd-maven-plugin-0.1.1.jar
      GOCD_PLUGIN_nessus-scan: https://github.com/Haufe-Lexware/gocd-plugins/releases/download/v1.0.0-beta/gocd-nessus-scan-plugin-1.0.0.jar
      GOCD_PLUGIN_s3-fetch: https://github.com/indix/gocd-s3-artifacts/releases/download/v2.0.2/s3fetch-assembly-2.0.2.jar
      GOCD_PLUGIN_s3-publish: https://github.com/indix/gocd-s3-artifacts/releases/download/v2.0.2/s3publish-assembly-2.0.2.jar
      GOCD_PLUGIN_script-executor: https://github.com/gocd-contrib/script-executor-task/releases/download/0.3/script-executor-0.3.0.jar
      GOCD_PLUGIN_slack: https://github.com/Vincit/gocd-slack-task/releases/download/v1.3.1/gocd-slack-task-1.3.1.jar
      GOCD_PLUGIN_sonar: https://github.com/Haufe-Lexware/gocd-plugins/releases/download/v1.0.0-beta/gocd-sonar-qualitygates-plugin-1.0.0.jar
      GOCD_USER_admin: '|(O>vF}4]/ic,+;?'
    volumes:
    - gocd-server-data:/data
    tty: true
    labels:
      io.rancher.scheduler.affinity:container_label_soft_ne: io.rancher.stack_service.name=$${stack_name}/$${service_name}
      io.rancher.container.hostname_override: container_name
  lb:
    image: rancher/lb-service-haproxy:v0.6.2
    ports:
    - 8153:8153/tcp
    labels:
      io.rancher.container.agent.role: environmentAdmin
      io.rancher.container.create_agent: 'true'
  gocd-agent:
    image: webcenter/alpine-gocd-agent:17.3.0-2
    environment:
      DOCKER_HOST: docker-engine:2375
      GOCD_CONFIG_agent_key: uE%th80B,Q0|4VPK
      GOCD_CONFIG_agent_resource_docker: docker
      GOCD_CONFIG_memory: 2048m
    volumes:
    - gocd-agent-data:/data
    - gocd-scheduler-setting:/opt/scheduler
    tty: true
    links:
    - gocd-server:gocd-server
    labels:
      io.rancher.container.agent.role: environment
      io.rancher.scheduler.affinity:container_label_soft_ne: io.rancher.stack_service.name=$${stack_name}/$${service_name}
      io.rancher.sidekicks: docker-engine,rancher-cattle-metadata
      io.rancher.container.create_agent: 'true'
      io.rancher.container.hostname_override: container_name
```

### Rancher Compose
```
version: '2'
services:
  docker-engine:
    retain_ip: true
    scale: 1
    start_on_create: true
  rancher-cattle-metadata:
    retain_ip: true
    scale: 1
    start_on_create: true
  gocd-server:
    scale: 1
    start_on_create: true
    health_check:
      healthy_threshold: 2
      response_timeout: 5000
      port: 8153
      unhealthy_threshold: 3
      interval: 5000
      strategy: recreate
  lb:
    scale: 1
    start_on_create: true
    lb_config:
      certs: []
      port_rules:
      - priority: 1
        protocol: http
        service: gocd-server
        source_port: 8153
        target_port: 8153
    health_check:
      healthy_threshold: 2
      response_timeout: 2000
      port: 42
      unhealthy_threshold: 3
      interval: 2000
      strategy: recreate
  gocd-agent:
    retain_ip: true
    scale: 1
    start_on_create: true
```

**Comment by disaster37** [comment]
The problem is your admin password. I have fixed the problem on 17.7.0-1 version (add double quote on htpassd command line).

You can use my catalog, it's wehere you can found last release of GoCD: `https://github.com/disaster37/webcenter-catalog.git` with the branch `master`.

**Comment by yonas** [comment]
Using your catalog works, thanks! :)
